external/kafka-newapi/src/main/scala/org/apache/spark/streaming/kafka/newapi/KafkaRDD.scala

```diff
@@ -169,7 +169,10 @@ class KafkaRDD[K: ClassTag, V: ClassTag, R: ClassTag] private[spark] (
 
     private def fetchBatch: Iterator[ConsumerRecord[K, V]] = {
       consumer.seek(new TopicPartition(part.topic, part.partition), requestOffset)
-      val recs = consumer.poll(pollTime)
+      var recs: ConsumerRecords[K, V] = null
+      do {
+        recs = consumer.poll(pollTime)
+      } while (recs.isEmpty && requestOffset < part.untilOffset)
```

Comment:
I think the removal of recursive call is right. 

>>
make sure this works when the topic is empty
<<
the '&& requestOffset < part.untilOffset' catches that i think